java/vector/src/main/java/org/apache/arrow/vector/types/Types.java

```diff
@@ -115,8 +115,8 @@
   public enum MinorType {
     NULL(Null.INSTANCE) {
       @Override
-      public Field getField() {
-        return NULL_FIELD;
+      public Field getField(String name) {
+        return NULL_FIELD.newFieldWithName(name);
```

Comment:
The minor type is derived from the Field object. 
We should be able to just pass the Field along wherever we pass the minor type and use that instead of trying to reconstruct it from the MinorType. The minorType can always be derived from Field when needed.